tap_shiphero/sync.py

```diff
@@ -38,10 +38,34 @@ def get_bookmark(state, stream_id, default):
         .get(stream_id, default)
     )
 
-def set_bookmark(state, stream_id, value):
+def set_bookmark(state, stream_id, field, value):
+    """Structure of the state dictionary:
+    {
+      "bookmarks": {
+        "stream1" : {
+          "datetime": "2019-01-01T00:00:00Z",
+          "page": 75
+        },
+        "stream2" : {
+          "datetime": "2019-01-01T00:00:00Z",
+          "page": 75
+        },
+        ...
+      }
+      ...
+    }
+    """
     if 'bookmarks' not in state:
         state['bookmarks'] = {}
-    state['bookmarks'][stream_id] = value
+
+    # This is the old style, so convert it to a dict
+    if isinstance(state['bookmarks'][stream_id], string):
+        state['bookmarks'][stream_id] = {}
+
+    if not isinstance(state['bookmarks'][stream_id], dict):
+        raise Exception('Unexpected bookmark format')
```

Comment:
I don't have a strong opinion of whether this should be included. I don't think it should ever be anything other than a string or a dict